### PR TITLE
Add augmentation option

### PR DESCRIPTION
**Pull request by keigohtr**
## What is this PR for?

Add augmentation technique to generalize the model.
Now we can use `ImageDataGenerator`-like class as preprocessor.

## This PR includes

- Renew `preprocessor.py`

## What type of PR is it?

Feature

## What is the issue?

https://github.com/abeja-inc/platform-template-image-classification/issues/6
https://github.com/abeja-inc/platform-template-image-classification/issues/7
https://github.com/abeja-inc/platform-template-image-classification/issues/9

## How should this be tested?

N/A

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ This template uses transfer-learning from VGG16 ImageNet. Making it easy to try
 | --- | --- | --- |
 | BATCH_SIZE | int | Batch size. Default `32`. |
 | EPOCHS | int | Epoch number. This template applies "Early stopping". Default `50`. |
-| LEARNING_RATE | float | Learning rate. Need to be from `0.0` to `1.0`. Default `0.001`. |
+| LEARNING_RATE | float | Learning rate. Need to be from `0.0` to `1.0`. Default `0.0001`. |
```
Comment (keigohtr) [inline README.md:22]: Change default value of `LEARNING_RATE` since a small learning rate may be good for a transfer learning.

File: README.md [hunk 2]
```diff
@@ -28,13 +28,27 @@ This template uses transfer-learning from VGG16 ImageNet. Making it easy to try
 | DROPOUT_SEED | int | Random seed for Dropout. Default `42`. |
 | EARLY_STOPPING_TEST_SIZE | float | Test data size for "Early stopping". Need to be from `0.0` to `1.0`. Default `0.2`. |
 | EARLY_STOPPING_PATIENCE | int | Number of patience for "Early stopping". Default `5`. |
-| IMG_ROWS | int | Image rows. Automatically resize to this size. Default `128`. |
-| IMG_COLS | int | Image cols. Automatically resize to this size. Default `128`. |
+| IMG_ROWS | int | Image rows. Automatically resize to this size. Default `224`. |
+| IMG_COLS | int | Image cols. Automatically resize to this size. Default `224`. |
```
Comment (keigohtr) [inline README.md:32]: Pretrained VGG16 uses `224`.

File: preprocessor.py [hunk 3]
```diff
@@ -1,11 +1,299 @@
...
-from keras.preprocessing.image import img_to_array
+from keras.preprocessing.image import (img_to_array,
+                                       apply_affine_transform,
+                                       apply_channel_shift,
+                                       apply_brightness_shift)
+
+
+class PreProcessor(object):
+    """PreProcessor
+    Referred from keras ImageDataGenerator.
```
Comment (keigohtr) [inline preprocessor.py:15]: This `PreProcessor` is inspired from `ImageDataGenerator`.
https://github.com/keras-team/keras/blob/master/keras/preprocessing/image.py#L233

**Review by keigohtr — COMMENTED** [review]
I made comments to my major changes.

File: utils/data_generator.py [hunk 4]
```diff
@@ -56,7 +68,7 @@ def __data_load(self, imgs, labels, start_pos: int, from_i: int, to_i: int):
                 file_content = source_data.get_content(cache=USE_CACHE)
                 file_like_object = io.BytesIO(file_content)
                 img = load_img(file_like_object, target_size=(IMG_ROWS, IMG_COLS))
-                img = preprocessor(img)
+                img = preprocessor.transform(img, is_train=self.is_train, seed=RANDOM_SEED)
```
Comment (ysku) [inline utils/data_generator.py]: seems better to do this after if-else clause because it is common in each condition.

**Review by ysku — COMMENTED** [review]
left just one comment 🙏

File: utils/data_generator.py (hunk 4 above)
Comment (keigohtr) [reply utils/data_generator.py]: Thank you!

**Review by ysku — APPROVED** [review]
LGTM 👍
